### ISSUE TITLE
Data encryption for Changefeeds

### ISSUE THREAD

**Issue by chriscasano** [cockroachdb/cockroach#45310]
This is a request to have encryption for data that is generated by Changefeed before sending to its sink.  An example of this, would be using AWS client side encryption: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingClientSideEncryption.html. The problem resides for sensitive data sets that need an extra layer of secure transfer and at-rest.   It would be nice to have a standard encryption process that can work across most of the Changefeed sinks where the use can provide the master key for encryption, similar to how we do encrypted backups.

**Comment by BramGruneir** [comment]
Assigning to @mwang1026 for triage.

**Comment by mwang1026** [comment]
We've triaged but want to wait until committing until we get confirmation on what encryption protocols to support (since presumably the data would be encrypted outside of CRDB).

**Comment by BramGruneir** [comment]
@thtruo for visibility

**Comment by chriscasano** [comment]
@mwang1026 probably worth considering this for backups too and not just changefeeds.

**Comment by mwang1026** [comment]
Yup. What are some examples of encryption protocols people are asking for? There are likely two parts to this--generic encryption UX and framework, and the incremental work to implement each encryption protocol.

**Comment by BramGruneir** [comment]
@mwang1026, any movement on this?  This looks like it could use similar work being done with AWS' KMS for encrypted backups.

**Comment by mwang1026** [comment]
No movement. Are people not using changefeeds because of this?

**Comment by BramGruneir** [comment]
It came as a customer request:  https://cockroachdb.zendesk.com/agent/tickets/4785

**Comment by data-matt** [comment]
Open 
CC @amruss
